commons-db/src/test/java/com/palantir/nexus/db/ThreadConfinedProxyTest.java

```diff
@@ -245,6 +246,55 @@ public void run() {
         assertEquals(testString, Iterables.getOnlyElement(subject));
     }
 
+    @Test
+    public void testPropagateExceptions() throws SQLException {
```

Comment:
Thank for the tests!

One minor point, can we do these with the ExpectedException rule instead? Will clean them up a bit and will make it easier to assert some more complex properties of these.